mathesar_ui/src/stores/abstractTypes.ts

```diff
@@ -0,0 +1,177 @@
+import { derived, writable, get } from 'svelte/store';
+import { getAPI, States } from '@mathesar/utils/api';
+import { currentDBId } from '@mathesar/stores/databases';
+import { preloadCommonData } from '@mathesar/utils/preloadData';
+
+import type { Readable, Writable, Unsubscriber } from 'svelte/store';
+import type { Database, DbType, AbstractTypeResponse } from '@mathesar/App.d';
+import type { CancellablePromise } from '@mathesar/components';
+
+const commonData = preloadCommonData();
+
+export interface AbstractType extends Omit<AbstractTypeResponse, 'db_types'> {
+  dbTypes: Set<DbType>,
+  // In the future, this would be base64 or link to svg. Currently it is just a direct string.
+  icon: string,
+  defaultDbType?: DbType,
+}
+
+export const UnknownAbstractType: AbstractType = {
+  name: 'Unknown',
+  identifier: 'unknown',
+  dbTypes: new Set(),
+  icon: '?',
+};
+
+export type AbstractTypeStoreData = Map<AbstractType['identifier'], AbstractType>;
+
+export interface AbstractTypeStore {
+  state: States,
+  data: AbstractTypeStoreData,
+  error?: string
+}
```

Comment:
Makes sense.. I have made some changes in https://github.com/centerofci/mathesar/pull/786/commits/f1dfed429c457130af0a48d64eeeca529cee5959.

I've introduced a pattern which calls the content of a store as Substance. Based on this:
* The type that implements `Writable` or `Readable` would be suffixed by `Store`.
* The value that is present within a store would be suffixed by `Substance`.

In this case, I've updated it to:
```typescript
export type AbstractTypesMap = Map<AbstractType['identifier'], AbstractType>;

interface AbstractTypesSubstance {
  state: States,
  data: AbstractTypesMap,
  error?: string
}

// Not in code, just for representational purpose
type AbstractTypesStore = Writable<AbstractTypesSubstance>;
```

Let me know if this looks good, or if we can come up with a better term instead of `Substance`.

Reply:
> calls the content of a store as Substance

I like it!